include/types/bookmarkObject.class.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * Copyright (C) 2013 Kolibre
+ *
+ * This file is part of Kolibre-KADOS.
+ * Kolibre-KADOS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2.1 of the License, or
+ * at your option) any later version.
+ *
+ * Kolibre-KADOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Kolibre-KADOS. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+require_once('AbstractType.class.php');
+require_once('bookmarkSet.class.php');
+
+
+class bookmarkObject extends AbstractType {
+
+   
+
+    /**
+     * @var (object)bookmarkSet
+     */
+    public $bookmarkSet;
+
+     /**
+     * @var string with format YYYY-MM-DDThh:mm:ss+hh:mm or YYYY-MM-DDThh:mm:ssZ
+     */
+    public $lastModifiedDate;
+
+    /******************** public functions ********************/
+
+    /**
+     * constructor for class bookmarkObject
+     */
+    function __construct($_bookmarkSet = NULL,$_lastModifiedDate = NULL) {
+        if (is_string($_bookmarkSet)) $this->setBookmarkSet($_bookmarkSet);
```

Comment:
$_bookmarkSet is not a string and the setBookmarkSet function will never be invoked in the constructor when $_bookmarkSet is an object of type 'bookmarkSet'. You should instead use ```if(is_a($_bookmarkSet, 'bookmarkSet'))```